codegenerator/cli/src/cli_args/interactive_init/validation.rs

```diff
@@ -77,12 +78,26 @@ impl<T> UniqueValueValidator<T> {
 
 pub fn is_not_empty_string_validator(s: &str) -> Result<Validation, CustomUserError> {
     if s.trim().is_empty() {
-        Ok(Validation::Invalid("Invalid empty string input".into()))
+        Ok(Validation::Invalid("Invalid input of empty string".into()))
     } else {
         Ok(Validation::Valid)
     }
 }
 
+// must start with a letter
+// only contain lowercase letters, numbers or underscorse
+// must have a maximum length of 63 characters
+pub fn is_valid_project_name_validator(s: &str) -> Result<Validation, CustomUserError> {
+    if is_valid_postgres_db_name(s) && is_valid_npm_package_name(s) {
```

Comment:
Note we actually put the name into the generated package.json so it also should be a valid package name.

Reply:
@DZakh why does it need to be a valid postgres name? Is that for the hosted service?

Reply:
> Note we actually put the name into the generated package.json so it also should be a valid package name.

Not necessarily, it won't break anything if it's invalid there, unless it doesn't break JSON

Reply:
> why does it need to be a valid postgres name

I don't know. As I remember we use repo name for the projects